docs/decisions/0005-optional-import-of-confluent-kafka.rst

```diff
@@ -0,0 +1,60 @@
+4. Optional import of confluent-kafka
+#####################################
+
+Status
+******
+
+Accepted
+
+Context
+*******
+
+* `confluent-kafka`_ is a library written and maintained by Confluent, our managed instance provider (see :doc:`0004-kafka-managed-hosting`). The library abstracts out the work for sending and receiving events to and from the Kafka cluster and converting them into message objects.
+* confluent-kafka in turn is a wrapper around a C library called `librdkafka`_ (distributed as `librdkafka_dev`)
+* librdkafka-dev does not currently have a compiled binary available for Linux/aarch64
+
+As a result of the points above, if a package includes a dependency on confluent-kafka, developers will not be able to install the package in Linux/aarch64 environments.
+
+.. _confluent-kafka: https://github.com/confluentinc/confluent-kafka-python
+.. _librdkafka: https://github.com/edenhill/librdkafka
+
+Decision
+********
+
+Instead of requiring confluent-kafka directly in base.in, we will wrap all imports of `confluent_kafka`in a `try...catch` block. If the import fails, the library will log an informative message and any calls will fail gracefully.
+
+For example::
+
+    try:
+        import confluent_kafka
+        from confluent_kafka import DeserializingConsumer
+    except ImportError:
+        confluent_kafka = None
+
+Then, later on, before any usage of `DeserializingConsumer`::
+
+    if not confluent_kafka:
+        warn("Confluent_kafka not installed")
```

Comment:
Shouldn't we warn during import, if at all? Having lots of repeated warnings would be annoying for those that have no intent on using the feature. Or, is this only run by code that is clearly meant to be using the feature?

Reply:
I do like the idea that this only warns on import, but the examples below use this pattern where the warning shows up when we try to use `confluent_kafka`. I'll leave things the way it is now, and then make a decision about this a bit later.

Reply:
This is only run by code that tries to use the feature. My worry is that, depending on usage, the import warning might be far away from the failure that will happen when someone tries to actually use the code.